server/src/main/scala/org/scassandra/server/cqlmessages/types/ColumnType.scala

```diff
@@ -15,21 +15,26 @@
  */
 package org.scassandra.server.cqlmessages.types
 
-import akka.util.ByteIterator
+import akka.util.{ByteString, ByteIterator}
 import com.typesafe.scalalogging.LazyLogging
-import org.apache.cassandra.serializers.TypeSerializer
 import org.scassandra.cql._
 import org.scassandra.server.cqlmessages.ProtocolVersion
 
+object WriteScope extends Enumeration {
+  val Standalone, InCollection = Value
+}
+
 abstract class ColumnType[T](val code : Short, val stringRep: String) extends LazyLogging {
-  def readValue(byteIterator : ByteIterator, protocolVersion: ProtocolVersion) : Option[T]
-  def writeValue(value : Any) : Array[Byte]
-  def writeValueInCollection(value: Any) : Array[Byte] = ???
-  def readValueInCollection(byteIterator: ByteIterator) : T = ???
-  def serializer: TypeSerializer[T] = ???
-  def convertToCorrectCollectionTypeForList(list: Iterable[_]) : List[T] = ???
-  def convertToCorrectCollectionTypeForSet(set: Iterable[_]) : Set[T] = convertToCorrectCollectionTypeForList(set).toSet
-  def convertToCorrectJavaTypeForSerializer(value: Any): T
+  implicit val byteOrder = java.nio.ByteOrder.BIG_ENDIAN
+  def readValue(byteIterator : ByteIterator)(implicit protocolVersion: ProtocolVersion) : Option[T]
+  def writeValue(value : Any)(implicit protocolVersion: ProtocolVersion) : Array[Byte]
+  def readValueWithLength(byteIterator: ByteIterator)(implicit protocolVersion: ProtocolVersion): Option[T] = {
```

Comment:
Additionally, there's some inefficiencies I think i've introduced in code going back and forth between ByteIterator and Array[Byte].  I'll look to clean that up a bit.